docs/nydus-failover-upgrade.md

```diff
@@ -0,0 +1,67 @@
+# Nydus Failover and Hot Upgrade
+
+Currently, nydusd supports failover and hot upgrade under fusedev and fscache modes, specifically:
+
+- Nydusd can recover automatically from crashes. After nydusd crashes then recovers, it can still serve container I/O operations.
+- Users can upgrade the version of nydusd without affecting the normal operation of containers. After the old nydusd exits and the new nydusd starts, it can still serve container I/O operations
```

Comment:
Users can upgrade nydusd without disrupting container I/O operations, once the old nydusd process exits and the new one starts, it can immediately serve container I/O.